dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -66,24 +67,69 @@ struct ReadFromStreamResult
     RegionPtr region;
 };
 
+void PreHandlingTrace::waitForSubtaskResources(uint64_t region_id, size_t parallel)
+{
+    auto parallel_subtask_limit = kvstore->getMaxParallelPrehandleSize();
+    {
+        auto current = ongoing_prehandle_subtask_count.load();
+        if (likely(ongoing_prehandle_subtask_count.compare_exchange_weak(current, current + parallel)))
+        {
+            LOG_DEBUG(
+                log,
+                "Prehandle resource meet, limit={}, current={}, region_id={}",
+                parallel_subtask_limit,
+                ongoing_prehandle_subtask_count.load(),
+                region_id);
+            return;
+        }
+    }
```

Comment:
If `parallel_subtask_limit = 16`, `current = 15`, `parallel = 4`. It will make the `ongoing_prehandle_subtask_count` = `15 + 4 == 19` which exceed the limit `16`

Reply:
fixed